server/mediaserver.go

```diff
@@ -187,7 +194,30 @@ func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID str
 			mid, key = parseManifestID(resp.ManifestID), resp.StreamKey
 			// Process transcoding options presets
 			if len(resp.Presets) > 0 {
-				presets = parsePresets(resp.Presets)
+				profiles = parsePresets(resp.Presets)
+			}
+
+			if len(resp.Profiles) > 0 {
+				for _, profile := range resp.Profiles {
+					name := profile.Name
+					if name == "" {
+						profile.Name = "webhook_" + common.DefaultProfileName(
+							int(profile.Width),
+							int(profile.Height),
+							int(profile.Bitrate))
+					}
+					prof := ffmpeg.VideoProfile{
+						Name:       profile.Name,
+						Bitrate:    fmt.Sprintf("%dk", profile.Bitrate),
+						Framerate:  uint(profile.FPS),
+						Resolution: fmt.Sprintf("%dx%d", profile.Width, profile.Height),
+					}
+					profiles = append(profiles, prof)
+				}
+			}
+
+			if len(resp.Profiles) <= 0 && len(resp.Presets) <= 0 {
```

Comment:
probably we should check `len(profiles)` here? `parsePresets` can skip profile, so len(profiles) can be zero even if len(resp.Presets) is not zero

Reply:
Actually this was intentional. We only want to set defaults if the user did *not* specify a preset/profile. If the parsing of the preset/profile failed, then setting the default instead is surprising and not what the user intended. In that case, it seems better to continue with an empty set of profiles (so no transcoding happens until the user fixes their configuration). We probably should add a comment here explaining, "Only set defaults if user did not specify a preset/profile"

Reply:
That makes sense @j0sh ... I agree that default should only be set if the user does not profile profiles/presets. I'll add a comment to explain